crates/m3/src/builder/constraint_system.rs

```diff
@@ -182,20 +182,31 @@ impl<F: TowerField> ConstraintSystem<F> {
 			if count == 0 {
 				continue;
 			}
-			if table.is_power_of_two_sized() {
-				if !count.is_power_of_two() {
-					return Err(Error::TableSizePowerOfTwoRequired {
-						table_id: table.id,
-						size: count,
-					});
-				}
-				if count != 1 << table.log_capacity(count) {
-					panic!(
-						"Tables with required power-of-two size currently cannot have capacity \
+			match table.size_spec() {
+				TableSizeSpec::PowerOfTwo => {
+					if !count.is_power_of_two() {
+						return Err(Error::TableSizePowerOfTwoRequired {
+							table_id: table.id,
+							size: count,
+						});
+					}
+					if count != 1 << table.log_capacity(count) {
+						panic!(
+						    "Tables with required power-of-two size currently cannot have capacity \
 						exceeding their count. This is because the flushes do not have automatic \
 						selectors applied, and so the table would flush invalid events"
-					);
+					    );
+					}
 				}
+				TableSizeSpec::Fixed { log_size } => {
+					if count != 1 << log_size {
```

Comment:
Maybe it would make sense for table.log_capacity(count) to return log_size for fixed tables, and to somehow unify Fixed and PowerOfTwo. Just a thought.

Reply:
I don't see how we could unify Fixed and PowerOfTwo. There are some tables whose structured polynomials are fixed size and others (like for memory address) that are structured and dynamically sized.